File: Python/768.最多能完成排序的块 II.py
from typing import List

class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        ans: int = 0
        sum1, sum2 = 0, 0
        t: List[int] = arr.copy()
        t.sort()
        for i in range(len(arr)):
            sum1 += arr[i]
            sum2 += t[i]
            if sum1 == sum2:
                ans += 1
        return ans

File: 768.最多能完成排序的块 II.cc
#include <vector>
#include <algorithm>
using std::vector;
using std::sort;

class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        int ans = 0;
        long sum1 = 0, sum2 = 0;
        vector<int> t = arr;
        sort(t.begin(), t.end());
        for(int i = 0; i < arr.size(); i++) {
            sum1 += t[i];
            sum2 += arr[i];
            if(sum1 == sum2) ans++;
        }
        return ans;
    }
};
